server/server.go

```diff
@@ -876,6 +878,34 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	return &httpS
 }
 
+var extensionsPattern = regexp.MustCompile(`^extension(.*).js$`)
+
+func (a *ArgoCDServer) serveExtensions(extensionsSharedPath string, w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "application/javascript")
+
+	content := ""
+	if err := filepath.Walk(extensionsSharedPath, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && extensionsPattern.MatchString(info.Name()) {
+			data, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				return err
+			}
+			content = content + fmt.Sprintf(`
```

Comment:
Agree. Working on adding caching

Reply:
resolved accidentally, unresolved :)

Reply:
One downside to this vs. `http.FileServer` is that (I believe) that library streams the requested file from disk instead of loading it fully into memory. In the case of this new behavior, multiple files are loaded simultaneously into memory before being sent out.

If we can stream the concatenated files out instead of caching them, I think that would sufficiently limit memory load on the API server.

Reply:
done!